gobblin-metrics/src/main/java/gobblin/metrics/reporter/ScheduledReporter.java

```diff
@@ -97,6 +101,26 @@ public ScheduledReporter(String name, Config config) {
         ExecutorsUtils.newThreadFactory(Optional.of(log), Optional.of("metrics-" + name + "-scheduler")));
     this.reportingPeriodSeconds = parsePeriodToSeconds(
         config.hasPath(REPORTING_INTERVAL) ? config.getString(REPORTING_INTERVAL) : DEFAULT_REPORTING_INTERVAL_PERIOD);
+    this.metricFilter = createMetricFilter(this.config);
+  }
+
+  /**
+   * Create a {@link MetricFilter} from the config key {@link #METRIC_FILTER_CLASS}.
+   */
+  private MetricFilter createMetricFilter(Config config) {
```

Comment:
See https://docs.google.com/document/d/1fX4Xg4LW0svfkSq3OsnjByQ3vxaILwAxRIaaDDzM78I/edit#heading=h.frw91twb0jx0 (LinkedIn private) for the plan on metric filters. The idea is for the declaration of metric filters to be expressive, instead of the usual classes. 
Metric filters will be based on regular expressions on metric names and metric types. If we have a prefix for profiling metrics, then we could use `^(?!internal\.profiling\.).*`.
If doing the regex is too much work, we can leave it for later, but let's make sure whatever is implemented now does not make it hard or ugly to implement the plan later on.